### PR TITLE
Upgrade to Angular 7 and its dependencies

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
We're currently running ng6.

## What is the new behavior?
This upgrades to ng7. There are a few breaking changes as a result on underlying dependency modifications (karma runs tests in random order now). I've patched the existing tests after running the test suite several hundred times, but there may be a few outstanding that I didn't catch.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

### CODE REVIEW

File: libs/state/src/product/reducers/index.spec.ts [hunk 1]
```diff
@@ -26,12 +27,17 @@ describe('selectProductState', () => {
 
     mockProduct = productFactory.create();
     store = TestBed.get(Store);
+
+    store.pipe(
+      take(1)
+    ).subscribe(s => console.log(s));
```
Comment (lderrickable) [inline libs/state/src/product/reducers/index.spec.ts]: Looks like this should be removed

Comment (damienwebdev) [reply libs/state/src/product/reducers/index.spec.ts]: Fixed

File: apps/foundation-demo/src/polyfills.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@
 
 /** Evergreen browsers require these. **/
```
[the remark this answers is not in the document]
Comment (damienwebdev) [reply apps/foundation-demo/src/polyfills.ts]: fixed.

**Comment by damienwebdev** [comment]
@lderrickable I've rebased this off develop

**Pull request merged**